In [43]:
import pandas as pd

In [44]:
# Load dataset
df1 = pd.read_csv('covariates.csv')

In [45]:
# Group data by sex and calculate mean weight
grouped = df1.groupby('e3_sex_None')['hs_c_weight_None'].mean()

In [46]:
# Print result
print(grouped)

e3_sex_None
female    27.936513
male      29.026696
Name: hs_c_weight_None, dtype: float64


The "male" gender has the highest weight with a mean value of 29.026696

In [47]:
# Load remaining datasets
df2 = pd.read_csv('phenotype.csv')
df3 = pd.read_csv('codebook.csv')
df4 = pd.read_csv('exposome.csv')

In [48]:
#Print shape of the dataset
print(df2.shape)

(1301, 7)


In [49]:
#Print column names of the dataset
print(df2.columns)

Index(['ID', 'e3_bw', 'hs_asthma', 'hs_zbmi_who', 'hs_correct_raven',
       'hs_Gen_Tot', 'hs_bmi_c_cat'],
      dtype='object')


In [50]:
#Print the data types of the columns in the dataset
print(df2.dtypes)

ID                    int64
e3_bw                 int64
hs_asthma             int64
hs_zbmi_who         float64
hs_correct_raven      int64
hs_Gen_Tot          float64
hs_bmi_c_cat          int64
dtype: object


In [51]:
#Print first five rows
print(df2.head())


   ID  e3_bw  hs_asthma  hs_zbmi_who  hs_correct_raven  hs_Gen_Tot  \
0   1   4100          0         0.30                18        84.0   
1   2   4158          0         0.41                25        39.0   
2   3   4110          1         3.33                13        40.0   
3   4   3270          0        -0.76                28        54.5   
4   5   3950          0         0.98                19        18.0   

   hs_bmi_c_cat  
0             2  
1             2  
2             4  
3             2  
4             2  


In [52]:
#Print last five rows
print(df2.tail())

        ID  e3_bw  hs_asthma  hs_zbmi_who  hs_correct_raven  hs_Gen_Tot  \
1296  1297   2900          0         1.94                31     34.4375   
1297  1298   3420          0        -0.46                34     28.5625   
1298  1299   4068          1         1.04                22     13.0000   
1299  1300   4000          0        -1.18                32     18.3125   
1300  1301   3580          0         0.48                28      0.0000   

      hs_bmi_c_cat  
1296             3  
1297             2  
1298             3  
1299             2  
1300             2  


In [53]:
# Print summary statistics phenotype
print(df2.describe())

               ID        e3_bw    hs_asthma  hs_zbmi_who  hs_correct_raven  \
count  1301.00000  1301.000000  1301.000000  1301.000000       1301.000000   
mean    651.00000  3389.302075     0.109147     0.403236         26.288240   
std     375.71066   509.994539     0.311943     1.189894          6.444849   
min       1.00000  1100.000000     0.000000    -3.580000          9.000000   
25%     326.00000  3080.000000     0.000000    -0.400000         22.000000   
50%     651.00000  3398.000000     0.000000     0.280000         27.000000   
75%     976.00000  3720.000000     0.000000     1.130000         32.000000   
max    1301.00000  5260.000000     1.000000     4.720000         36.000000   

        hs_Gen_Tot  hs_bmi_c_cat  
count  1301.000000   1301.000000  
mean     24.376722      2.385857  
std      19.028232      0.677268  
min       0.000000      1.000000  
25%      10.000000      2.000000  
50%      20.000000      2.000000  
75%      33.437500      3.000000  
max     133.00000

In [54]:
print(df3.shape)

(241, 13)


In [55]:
print(df3.columns)

Index(['Unnamed: 0', 'variable_name', 'domain', 'family', 'subfamily',
       'period', 'location', 'period_postnatal', 'description', 'var_type',
       'transformation', 'labels', 'labelsshort'],
      dtype='object')


In [56]:
print(df3.dtypes)

Unnamed: 0          object
variable_name       object
domain              object
family              object
subfamily           object
period              object
location            object
period_postnatal    object
description         object
var_type            object
transformation      object
labels              object
labelsshort         object
dtype: object


In [57]:
print(df3.head())
print(df3.tail())

               Unnamed: 0           variable_name             domain  \
0    h_abs_ratio_preg_Log    h_abs_ratio_preg_Log  Outdoor exposures   
1    h_no2_ratio_preg_Log    h_no2_ratio_preg_Log  Outdoor exposures   
2  h_pm10_ratio_preg_None  h_pm10_ratio_preg_None  Outdoor exposures   
3  h_pm25_ratio_preg_None  h_pm25_ratio_preg_None  Outdoor exposures   
4      hs_no2_dy_hs_h_Log      hs_no2_dy_hs_h_Log  Outdoor exposures   

          family     subfamily     period location        period_postnatal  \
0  Air Pollution  PMAbsorbance  Pregnancy     Home                     NaN   
1  Air Pollution           NO2  Pregnancy     Home                     NaN   
2  Air Pollution          PM10  Pregnancy     Home                     NaN   
3  Air Pollution         PM2.5  Pregnancy     Home                     NaN   
4  Air Pollution           NO2  Postnatal     Home  Day before examination   

                                         description var_type  \
0  abs value (extrapolated back i

In [58]:
print(df3.describe())

                  Unnamed: 0         variable_name     domain     family  \
count                    241                   241        241        241   
unique                   241                   241          6         20   
top     h_abs_ratio_preg_Log  h_abs_ratio_preg_Log  Chemicals  Lifestyle   
freq                       1                     1        106         39   

       subfamily     period location        period_postnatal  \
count        241        241       77                      21   
unique        81          2        2                       4   
top         Diet  Postnatal     Home  Day before examination   
freq          29        142       66                       7   

                                              description var_type  \
count                                                 241      241   
unique                                                240        2   
top     Dimethyl thiophosphate (DMTP) in child adjuste...  numeric   
freq              

In [59]:
#Count missing values per column
missing_values_count = df2.isnull().sum()
print(missing_values_count)

ID                  0
e3_bw               0
hs_asthma           0
hs_zbmi_who         0
hs_correct_raven    0
hs_Gen_Tot          0
hs_bmi_c_cat        0
dtype: int64


In [60]:
#Extract all IDs of indivuduals with a BMI classification at age 6-11 of overweight or obese
ids = df2[df2["hs_bmi_c_cat"].isin([3, 4])]["ID"].tolist()

# Print the IDs
print(ids)

[3, 10, 11, 12, 15, 16, 21, 25, 27, 30, 34, 39, 41, 43, 52, 54, 63, 67, 70, 75, 76, 77, 78, 79, 80, 81, 83, 84, 85, 87, 88, 95, 110, 112, 120, 121, 127, 140, 141, 142, 143, 144, 147, 150, 155, 159, 160, 170, 172, 174, 181, 183, 192, 193, 199, 200, 204, 206, 209, 213, 214, 225, 227, 229, 233, 247, 257, 259, 262, 266, 268, 270, 279, 284, 285, 292, 295, 297, 298, 299, 301, 302, 303, 304, 306, 309, 312, 315, 323, 324, 325, 328, 333, 336, 339, 340, 344, 345, 346, 351, 352, 359, 361, 362, 364, 366, 367, 379, 381, 385, 386, 388, 392, 397, 404, 407, 409, 410, 421, 424, 431, 433, 436, 437, 438, 441, 442, 443, 444, 447, 448, 452, 453, 458, 459, 461, 466, 469, 475, 481, 484, 490, 494, 497, 501, 502, 510, 512, 522, 523, 526, 530, 536, 540, 547, 550, 551, 552, 553, 554, 559, 562, 565, 574, 575, 577, 582, 584, 598, 599, 608, 609, 612, 615, 616, 617, 618, 620, 622, 623, 629, 631, 637, 638, 641, 645, 647, 648, 663, 664, 666, 670, 671, 680, 684, 686, 688, 690, 691, 702, 704, 707, 710, 712, 714, 717, 71

In [61]:
from collections import Counter

#Count occurences of each ID in the list
id_counts = Counter(ids)

#Print the IDs that occur more than once
duplicates = [id for id, count in id_counts.items() if count > 1]
print("The following IDs occur more than once in the extracted target cohort: ", duplicates)

The following IDs occur more than once in the extracted target cohort:  []


In [62]:
#Create a new dataframe with the cohort records
cohort_df = df[df["ID"].isin(ids)]

print(cohort_df.head())
print(cohort_df.tail())

    ID  h_cohort e3_sex_None  e3_yearbir_None  h_mbmi_None  hs_wgtgain_None  \
2    3         4        male             2008    30.116213             11.0   
9   10         1      female             2008    40.333778              9.0   
10  11         1      female             2008    32.073063             10.0   
11  12         3        male             2006    23.738663             18.0   
14  15         4        male             2008    43.027718             12.0   

    e3_gac_None  h_age_None  h_edumc_None  h_native_None  h_parity_None  \
2     39.000000   34.232422             3              2              1   
9     41.000000   25.000000             1              1              2   
10    40.714287   31.000000             3              0              1   
11    39.000000   37.598904             3              2              1   
14    41.428570   30.674887             2              2              0   

    hs_child_age_None  hs_c_height_None  hs_c_weight_None  
2            6

In [63]:
#Check for duplicate rows
if cohort_df.duplicated().any():
    print("There are duplicate rows in the cohort dataframe.")
else:
    print("There are no duplicate rows in the new dataframe.")

There are no duplicate rows in the new dataframe.


In [64]:
#Find suplicate rows across all columns except the ID column
similarRows = cohort_df[cohort_df.duplicated(subset=cohort_df.columns.difference(['ID']))]["ID"].tolist()

print("IDs for the rows that have duplicates except ID: ", similarRows)

IDs for the rows that have duplicates except ID:  []


In [68]:
# Intentionally duplicate the first 3 rows of the dataframe
cohort_df = pd.concat([cohort_df, cohort_df.iloc[:3]], ignore_index=True)



In [71]:
#Check for duplicate rows
if cohort_df.duplicated().any():
    print("There are duplicate rows in the cohort dataframe.")
else:
    print("There are no duplicate rows in the new dataframe.")

There are duplicate rows in the cohort dataframe.


In [72]:
#Find suplicate rows across all columns except the ID column
similarRows = cohort_df[cohort_df.duplicated(subset=cohort_df.columns.difference(['ID']))]["ID"].tolist()

print("IDs for the rows that have duplicates except ID: ", similarRows)

IDs for the rows that have duplicates except ID:  [3, 10, 11, 3, 10, 11]


In [73]:
# Reverse the intentional duplication
cohort_df = cohort_df.drop_duplicates(subset=cohort_df.columns.difference(['ID']), keep='first')

In [75]:
#Check for duplicate rows
if cohort_df.duplicated().any():
    print("There are duplicate rows in the cohort dataframe.")
else:
    print("There are no duplicate rows in the new dataframe.")

There are no duplicate rows in the new dataframe.


In [77]:
#Find suplicate rows across all columns except the ID column
similarRows = cohort_df[cohort_df.duplicated(subset=cohort_df.columns.difference(['ID']))]["ID"].tolist()

print("IDs for the rows that have duplicates except ID: ", similarRows)

IDs for the rows that have duplicates except ID:  []
